addons/mail/models/mail_channel.py

```diff
@@ -929,20 +948,27 @@ def channel_fetch_preview(self):
         """ Return the last message of the given channels """
         if not self:
             return []
-        self._cr.execute("""
-            SELECT mail_channel_id AS id, MAX(mail_message_id) AS message_id
-            FROM mail_message_mail_channel_rel
-            WHERE mail_channel_id IN %s
-            GROUP BY mail_channel_id
-            """, (tuple(self.ids),))
-        channels_preview = dict((r['message_id'], r) for r in self._cr.dictfetchall())
+        channels_last_message_ids = self._channel_last_message_ids()
+        channels_preview = dict((r['message_id'], r) for r in channels_last_message_ids)
```

Comment:
If all calls of "_channel_last_message_ids" make this dict convert, it should be the return value of the method to avoid copy/paste.

Reply:
it's not exactly the same, one it message_id : result and the other is channel_id: message_id

Reply:
It's strange for this one to index the dict by the message_id instead of channel_id but ok then.

Reply:
It's because it needs to call `message_format` on all messages to get previews, so it's a convenient way to list all needed messages I suppose. It could probably be refactored